connectors/sources/jira.py

```diff
@@ -0,0 +1,493 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Jira source module responsible to fetch documents from Jira on-prem or cloud server.
+"""
+import asyncio
+import os
+from copy import copy
+from datetime import datetime
+from functools import partial
+from urllib import parse
+
+import aiofiles
+import aiohttp
+import pytz
+from aiofiles.os import remove
+from aiofiles.tempfile import NamedTemporaryFile
+from aiohttp.client_exceptions import ServerDisconnectedError
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource
+from connectors.utils import (
+    TIKA_SUPPORTED_FILETYPES,
+    CancellableSleeps,
+    ConcurrentTasks,
+    MemQueue,
+    convert_to_b64,
+    iso_utc,
+    ssl_context,
+)
+
+RETRY_INTERVAL = 2
+FILE_SIZE_LIMIT = 10485760
+
+FETCH_SIZE = 100
+CHUNK_SIZE = 1024
+QUEUE_SIZE = 1024
+QUEUE_MEM_SIZE = 5
```

Comment:
I see you are multiplying this value by 1024*1024 then it is worth mentioning that the value is in megabytes. Another option is to move the multiplication to this line from line 86.

Reply:
I'll mention that the value is in megabytes through an inline comment.